orte/mca/rml/ofi/rml_ofi_component.c

```diff
@@ -1086,3 +1248,253 @@ static void orte_rml_ofi_fini(void *mod)
                         " %s -  orte_rml_ofi_fini() completed ",
                         ORTE_NAME_PRINT(ORTE_PROC_MY_NAME));
 }
+
+
+static void pr_cons(orte_rml_ofi_peer_t *ptr)
+{
+    ptr->socket_ep = NULL;
+    ptr->socket_ep_len = 0;
+    ptr->psmx_ep = NULL;
+    ptr->psmx_ep_len = 0;
+}
+static void pr_des(orte_rml_ofi_peer_t *ptr)
+{
+    if ( 0 < ptr->socket_ep_len)
+        free( ptr->socket_ep);
+    if ( NULL != ptr->psmx_ep )
+        free( ptr->psmx_ep);
+}
+OBJ_CLASS_INSTANCE(orte_rml_ofi_peer_t,
+                   opal_object_t,
+                   pr_cons, pr_des);
+
+
+/* The returned string will be of format - "<process-name>;ofi-socket:<addr_format,ip,portaddr>;ofi-<provider2>:<prov2epname>" */
+static char* ofi_get_contact_info(void)
+{
+    char *turi, *final=NULL, *tmp, *addrtype;
+    size_t len = 0;
+    int rc=ORTE_SUCCESS, cur_ofi_prov=0;
+    bool one_added = false;
+    struct sockaddr_in* ep_sockaddr;
+    
+    /* start with our process name */
+    if (ORTE_SUCCESS != (rc = orte_util_convert_process_name_to_string(&final, ORTE_PROC_MY_NAME))) {
+        /* [TODO] ORTE_ERROR_LOG(rc);*/
+        return final;
+    }
+    len = strlen(final);
+
+    /* The returned string will be of format - "<process-name>;ofi-socket:<sin_family,sin_addr,sin_port>;ofi-<provider2>:<prov2epname>" */
+    for( cur_ofi_prov=0; cur_ofi_prov < orte_rml_ofi.ofi_prov_open_num ; cur_ofi_prov++ ) {
+            switch ( orte_rml_ofi.ofi_prov[cur_ofi_prov].fabric_info->addr_format) {
```

Comment:
Same comment as above: if the `fi_av_straddr()` could be used, that would be less code to have here, and you won't need to know/understand all addressing formats.

I see similar constructs later -- I'll stop mentioning it here in the review, but see if `fi_av_straddr()` can  be used to avoid these kinds of things.